src/main/java/org/terasology/launcher/settings/LauncherSettingsValidator.java

```diff
@@ -29,19 +29,37 @@
 public final class LauncherSettingsValidator {
     private static final Logger logger = LoggerFactory.getLogger(LauncherSettingsValidator.class);
 
+    private static String removeUnsupportedJvmParameters(final String currentParams) {
+        String params = currentParams;
+
+        final String[] deprecatedParams = {"-XX:\\+UseParNewGC", "-XX:\\+UseConcMarkSweepGC", "-XX:ParallelGCThreads=10"};
+
+        for (String deprecatedParam : deprecatedParams) {
+            params = params.replaceAll(deprecatedParam, "");
+        }
+
+        return params;
+    }
+
     private static final List<SettingsValidationRule> RULES = Arrays.asList(
             // Rule for max heap size
             new SettingsValidationRule(
-                s -> !(System.getProperty("os.arch").equals("x86") && s.getMaxHeapSize().compareTo(JavaHeapSize.GB_1_5) > 0),
-                "Max heap size cannot be greater than 1.5 GB for a 32-bit JVM",
-                s -> s.setMaxHeapSize(JavaHeapSize.GB_1_5)
+                    s -> !(System.getProperty("os.arch").equals("x86") && s.getMaxHeapSize().compareTo(JavaHeapSize.GB_1_5) > 0),
+                    "Max heap size cannot be greater than 1.5 GB for a 32-bit JVM",
+                    s -> s.setMaxHeapSize(JavaHeapSize.GB_1_5)
             ),
 
             // Rule for initial heap size
             new SettingsValidationRule(
-                s -> s.getInitialHeapSize().compareTo(s.getMaxHeapSize()) < 0,
+                    s -> s.getInitialHeapSize().compareTo(s.getMaxHeapSize()) < 0,
                     "Initial heap size cannot be greater than max heap size",
-                s -> s.setInitialHeapSize(s.getMaxHeapSize())
+                    s -> s.setInitialHeapSize(s.getMaxHeapSize())
+            ),
+
+            new SettingsValidationRule(
+                    s -> s.getUserJavaParameters().isEmpty(),
```

Comment:
by making this condition `isEmpty`, doesn't this declare it invalid if **any** parameters are present?

Reply:
Yes, but this is just a poor man's check to avoid multiple `indexOf(...) >= 0` here - this will just trigger the correction, which may in the do nothing...

Reply:
oh, we have Guava! Can we do something like Sets.intersection(x,y).isEmpty?